### PR TITLE
Added AWS Requirements to sudoadmin

### PR DESCRIPTION
**Pull request by OnGle**
added banner file to /var/lib/inithooks/sudoadmin/root_ssh_banner
adds match and commented out banner line to /etc/ssh/sshd_config at install
(NOTE does not remove on uninstall!)

### CODE REVIEW

**Comment by alonswartz** [comment]
A few issues:

1. Would be better to encapsulate all of this functionality in the sudoadmin script. It will be a lot easier to maintain and to understand.

2. Your comment on adding additional commands AFTER the match block is incorrect. From the manpage:

```
Match   Introduces a conditional block.  If all of the criteria on the
             Match line are satisfied, the keywords on the following lines
             override those set in the global section of the config file,
             until either another Match line or the end of the file.
```

**Comment by OnGle** [comment]
I absolutely agree, I wasn't sure on how to proceed here, I've never dealt with debian packaging from this side and didn't know how the best way to move forward was.

I was aware it functioned this way. The way my sed lines check for Match lines, I was worried that duplicate "Match user root" lines would mess it up. However after reading over it I can understand how it doesn't make a whole lot of sense to allow users to add commands to a conditional block to be matched by regex, that's just asking for trouble.

Thanks for your feedback, I'll tidy it up a bit :)

**Comment by OnGle** [comment]
Ok I've just changed things so all the functionality is done in turnkey-sudoadmin and it should auto add nonexistent banner files as well as a nonexistent sudoadmin block silently. The comments have been changed to give a more general but understandable warning as well as the match is now encompassed within comment blocks (used as pattern ranges in sed) which restrict the regex from finding other match unrelated match lines.

**Comment by OnGle** [comment]
Also closes https://github.com/turnkeylinux/tracker/issues/541

File: turnkey-sudoadmin [hunk 1]
```diff
@@ -205,6 +236,7 @@ case $1 in
         permitrootlogin_ssh "yes";
         user_state "root" "unlock";
         user_state "admin" "lock";
+        set_root_login_banner "";
```
Comment (JedMeister) [inline turnkey-sudoadmin]: IMO this should probably be:
````
set_root_login_banner "/dev/null";
````

**Comment by alonswartz** [comment]
So I thought about this a little more, and did some testing. I think I came up with a simple solution:

A. We update `common/conf/turnkey.d/sshd` to add the following to sshd_config
```
Match user "root"
Banner /root/.ssh/banner
Match user *
```
I've tested and sshd doesn't have issues pointing to a banner file if it doesn't exist.
I also added a closing Match so if users append any directives it won't be included in the root block.

B. Have sudoadmin write/blankout `/root/.ssh/banner` depending on the state.

Thoughts?

**Comment by OnGle** [comment]
I think that's a totally reasonable solution, very clean too (I think I have too much fun with regex sometimes). Doing it this way means no potential issues with wild regex in edge cases, reduces the chances of users breaking the config. I also think `/root/.ssh/banner` is a more logical place for the banner, that way it's pretty explicit to what it's purpose is without having to make a descriptive title.

So remove the comment/uncomment functionality and replace with an `echo "$banner" > /root/.ssh/banner` and perhaps parse the banner as the argument to the function since it's a relatively small which would allow us to cut back the function to:

Assert banner block is in sshd_config
and overwrite banner file with the argument to the function which will either be "" or the "<The banner>".

I'll try setting that up today :)

**Comment by JedMeister** [comment]
That looks awesome to me! :+1: What do you think @alonswartz ?

**Comment by OnGle** [comment]
Whoops

**Comment by alonswartz** [comment]
Not exactly what I was referring to in https://github.com/turnkeylinux/inithooks/pull/21#issuecomment-185705127 but I'll merge and tweak.

**Comment by alonswartz** [comment]
For reference: https://github.com/turnkeylinux/inithooks/commit/9a4578835f29cc34ba963ea3b08b82c17d9e4384 and https://github.com/turnkeylinux/common/commit/e54741d7ddd0ce2cc1983fbf214776294bb7dda4

**Comment by OnGle** [comment]
Oh I understand, that's incredibly clean. Sorry for the misunderstanding :)

**Pull request merged**